src/transformers/models/auto/tokenization_auto.py

```diff
@@ -769,7 +769,13 @@ def from_pretrained(cls, pretrained_model_name_or_path, *inputs, **kwargs):
                 tokenizer_auto_map = config.auto_map["AutoTokenizer"]
 
         has_remote_code = tokenizer_auto_map is not None
-        has_local_code = config_tokenizer_class is not None or type(config) in TOKENIZER_MAPPING
+        has_local_code = type(config) in TOKENIZER_MAPPING or (
+            config_tokenizer_class is not None
+            and (
+                tokenizer_class_from_name(config_tokenizer_class) is not None
+                or tokenizer_class_from_name(config_tokenizer_class + "Fast") is not None
+            )
+        )
```

Comment:
Comment for the reviewer here: Before this change, models with custom tokenizer code would usually set `config_tokenizer_class`, but this class would not be one of the recognized library classes. As a result, when `trust_remote_code=None` for those repos, we would incorrectly infer `has_local_code=True`, and then this code block would continue before hitting the `Tokenizer class does not exist or is not currently imported` error clause below.

This change ensures we only infer `has_local_code=True` when `config_tokenizer_class` is set **and** we actually recognize that class as one of the library classes.